dora/core/common/src/main/java/alluxio/wire/WorkerIdentity.java

```diff
@@ -118,6 +120,33 @@ public String toString() {
         Parsers.getParserOfVersion(mVersion).getVersionSpecificRepresentation(this));
   }
 
+  /**
+   * Construct from a workerid representation from registration.
+   *
+   * @param workerIdentityStr
+   * @return WorkerIdentity object
+   * @throws InvalidArgumentException
+   */
+  public static WorkerIdentity fromString(String workerIdentityStr)
```

Comment:
Move this method close to `toString` so that hopefully people will remember to keep the two implementations compatible when they change either of them in the future. Also please add a unit test to ensure the string representation generated by `toString` can be correctly parsed by `fromString`.